### PR TITLE
Add a Validate method to the Debug type

### PR DESCRIPTION
**Pull request by thomas-fossati**

### CODE REVIEW

File: debug_test.go [hunk 1]
```diff
@@ -5,40 +5,41 @@ package eat
...
 import (
 	"errors"
+	"fmt"
 	"testing"
 
 	"github.com/stretchr/testify/assert"
 )
 
-func TestDebug_Set(t *testing.T) {
+func TestDebug_Check(t *testing.T) {
```
Comment (thomas-fossati) [inline debug_test.go:14]:
```suggestion
func TestDebug_Validate(t *testing.T) {
```

File: debug.go [hunk 2]
```diff
@@ -14,30 +17,67 @@ debug-disable-type = &(
...
 	}
 
-	*d = Debug(val)
-
 	return nil
 }
+
+// MarshalCBOR CBOR encodes a valid Debug receiver
+func (d Debug) MarshalCBOR() ([]byte, error) {
+	if err := d.Validate(); err != nil {
```
Comment (setrofim) [inline debug.go:69]: Given that we're not validating on unmarshalling, does it make sense to validate on marshalling? If someone puts an invalid value in, and we accept it, would then not also expect to be able to get it out?

Comment (thomas-fossati) [reply debug.go:69]: Right, but Postel's principle is asymmetric :-)  (I'm applying that here.)

Comment (setrofim) [reply debug.go:69]: I don't think that makes sense here. That talks about complete systems, this is a component. We're not creating the data we're putting out here. The point of splitting out bounds checking into a separate `Validate()` is to allow this to be used in systems where invalid data may already be present.

E.g. imagine this is used in  a system that reads a token in, passes it around (possibly routing it based on some of its values), and then writes it out, all without modifying it in any way. As an engineer implementing this I would expect the library I'm using to either, a) not care about the out of bounds values within leaving it to me check if I wish, or b) to fail early on reading the value in.

Comment (thomas-fossati) [reply debug.go:69]: I'm not sure that Postel is not applicable to emitters of single protocol elements, as in this case, as long as this strategy is applied all over to construct the whole protocol bottom-up.

I also think your example could be handled with what is here already: it suffices to decode the raw token while keeping it in memory somewhere, operate on the its internalised structure and then fetch the raw bytes from memory and spit it out.

However, I buy the point that this effectively mixes a bit of policy with a mechanism in a way that the user of the library cannot control completely.  And that I don't like.  So I am OK with untangling the two :-)

**Review by setrofim — APPROVED** [review]

**Pull request merged**